content/providers/02-openai-compatible-providers/index.mdx

```diff
@@ -73,3 +72,161 @@ const { text } = await generateText({
+packages/example/
+|-- src/
+|   |-- example-chat-settings.ts      # Chat model types and settings
+|   |-- example-completion-settings.ts # Completion model types and settings
+|   |-- example-embedding-settings.ts  # Embedding model types and settings
+|   |-- example-provider.ts           # Main provider implementation
+|   |-- example-provider.test.ts      # Provider tests
+|   `-- index.ts                      # Public exports
+|-- package.json
+|-- tsconfig.json
+|-- tsup.config.ts                    # Build configuration
+`-- README.md
+```
+
+### Key Files
+
+1. **example-chat-settings.ts** - Define chat model IDs and settings:
+
+```ts
+import { OpenAICompatibleChatSettings } from '@ai-sdk/openai-compatible';
+
+export type ExampleChatModelId =
+  | 'example/chat-model-1'
+  | 'example/chat-model-2'
+  | (string & {});
+
+export interface ExampleChatSettings extends OpenAICompatibleChatSettings {}
+```
+
+2. **example-provider.ts** - Main provider implementation:
+
+```ts
+import { OpenAICompatibleChatLanguageModel } from '@ai-sdk/openai-compatible';
+import { loadApiKey } from '@ai-sdk/provider-utils';
+
+export interface ExampleProviderSettings {
+  apiKey?: string;
+  baseURL?: string;
+  headers?: Record<string, string>;
+}
+
+export function createExample(options: ExampleProviderSettings = {}) {
+  const baseURL = options.baseURL ?? 'https://api.example.com/v1';
+  const getHeaders = () => ({
+    Authorization: `Bearer ${loadApiKey({
+      apiKey: options.apiKey,
+      environmentVariableName: 'EXAMPLE_API_KEY',
+      description: 'Example API key',
+    })}`,
+    ...options.headers,
+  });
+
+  // Create provider instance with chat, completion, and embedding models
+  const provider = (modelId, settings = {}) => {
+    return new OpenAICompatibleChatLanguageModel(modelId, settings, {
+      provider: 'example.chat',
+      url: ({ path }) => `${baseURL}${path}`,
+      headers: getHeaders,
+    });
+  };
+
+  return provider;
+}
+
+// Export default instance
+export const example = createExample();
+```
+
+3. **index.ts** - Public exports:
+
+```ts
+export { createExample, example } from './example-provider';
+export type {
+  ExampleProvider,
+  ExampleProviderSettings,
+} from './example-provider';
+```
+
+4. **package.json** - Package configuration:
+
+```js
+{
+  "name": "@ai-sdk/example",
+  "version": "0.0.1",
+  "dependencies": {
+    "@ai-sdk/openai-compatible": "^0.0.7",
+    "@ai-sdk/provider": "^1.0.2",
+    "@ai-sdk/provider-utils": "^2.0.4"
+  }
+}
+```
+
+### Usage
+
+Once published, users can use your provider like this:
+
+```ts
+import { example } from '@ai-sdk/example';
+import { generateText } from 'ai';
+
+const { text } = await generateText({
+  model: example('example/chat-model-1'),
+  prompt: 'Hello, how are you?',
+});
+```
+
+This structure provides a clean, type-safe implementation that leverages the OpenAI Compatible package while maintaining consistency with the usage of other AI SDK providers.
```

Comment:
move into separate page at top of subsection (similar what we do under community providers)